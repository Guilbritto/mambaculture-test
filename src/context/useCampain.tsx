'use client'

import { getCampain } from '@/app/data'
import { Campain } from '@/types/campain'
import { Paginated } from '@/types/paginated'
import React, { ReactNode, createContext, useCallback, useContext, useEffect, useState } from 'react'

type CampainContextType = {
    campains: Paginated<Campain[]>
    fetchCampains: (page?: number) => Promise<void>
}

const CampainContext = createContext<CampainContextType | undefined>(
  undefined,
)

const CampainProvider = ({ children }: { children: ReactNode }) => {
    const [campains, setCampains] = useState<Paginated<Campain[]>>({} as Paginated<Campain[]>)

    const fetchCampains = useCallback(async (page=1) => {
      const campain = await getCampain(page);
      setCampains(campain ?? {} as Paginated<Campain[]>)
  }, [])

    useEffect(() => {
      if(campains.data) return
      fetchCampains()
    }, [campains.data, fetchCampains])

    return (
        <CampainContext.Provider
            value={{ campains, fetchCampains }}
        >
        {children}
        </CampainContext.Provider>
    )
}

const useCampain = () => {
  const campainContext = useContext(CampainContext)
  if (!campainContext) {
    throw new Error('useCampain must be used within a CampainProvider')
  }
  return campainContext
}
export { CampainContext, CampainProvider, useCampain }
