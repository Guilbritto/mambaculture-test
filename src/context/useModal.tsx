'use client'

import React, { ReactNode, createContext, useContext, useState } from 'react'

type ModalContextType = {
  isOpen: boolean
  renderModal?: ReactNode
  openModal: () => void
  closeModal: () => void
  render: (renderModal: ReactNode) => void
}

const ModalContext = createContext<ModalContextType | undefined>(
  undefined,
)

const ModalProvider = ({ children }: { children: ReactNode }) => {
  const [isOpen, setIsOpen] = useState(false)
  const [renderModal, setRenderModal] = useState<
    ReactNode | undefined
  >()

  const render = (renderPageModal: ReactNode) => {
    setRenderModal(renderPageModal)
  }
  
  const openModal = () => {
    setIsOpen(true)
  }

  const closeModal = () => {
    setIsOpen(false)
    render(<></>)
  }

  return (
    <ModalContext.Provider
      value={{ isOpen, renderModal, openModal, closeModal, render }}
    >
      {children}
    </ModalContext.Provider>
  )
}
const useModal = () => {
  const modalContext = useContext(ModalContext)
  if (!modalContext) {
    throw new Error('useModal must be used within a ModalProvider')
  }
  return modalContext
}
export { ModalContext, ModalProvider, useModal }
