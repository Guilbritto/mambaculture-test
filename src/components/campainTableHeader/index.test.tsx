import React from 'react';
import { render } from '@testing-library/react';
import {CampainTableHeader} from './'; 
import { ModalProvider } from '@/context/useModal';

describe('CampainTableHeader', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(
        <ModalProvider>
            <CampainTableHeader />
        </ModalProvider>
    );
        expect(asFragment()).toMatchSnapshot();
    });
});