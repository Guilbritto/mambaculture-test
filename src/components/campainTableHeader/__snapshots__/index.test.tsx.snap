// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampainTableHeader should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-2 px-2  border-b border-b-slate-200 pb-2 flex justify-between items-center mb-3 "
  >
    <h1
      class="text-xl font-semibold text-primary"
    >
       Painel de controle 
    </h1>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 rounded-lg bg-slate-700"
    >
      Criar campanha
    </button>
  </div>
</DocumentFragment>
`;
